frontend/apps/reader/modules/readermenu.lua

```diff
@@ -305,15 +306,19 @@ end
 
 function ReaderMenu:onSwipeShowMenu(ges)
     if self.activation_menu ~= "tap" and ges.direction == "south" then
-        self.ui:handleEvent(Event:new("ShowConfigMenu"))
+        if self.show_bottom_menu then
```

Comment:
readSettings() (and friends like nilOrTrue()) cost nothing, so you could just use there:
`if G_reader_settings:nilOrTrue("show_bottom_menu") then`
and you would not need to say "a restart is needed".